web/src/main/java/com/epam/esm/security/UserDetailServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package com.epam.esm.security;
+
+import com.epam.esm.entity.User;
+import com.epam.esm.repository.impl.UserRepositoryImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserDetailServiceImpl implements UserDetailsService {
+
+    private final UserRepositoryImpl userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userRepository.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User not found"));
```

Comment:
"User not found" - need add localization

Reply:
Done